spec/datadog/appsec/processor_spec.rb

```diff
@@ -0,0 +1,227 @@
+
+    logger = double(Datadog::Core::Logger)
+    allow(logger).to receive(:debug?).and_return true
+    allow(logger).to receive(:debug)
+    allow(logger).to receive(:info)
+    allow(logger).to receive(:warn)
+    allow(logger).to receive(:error)
+
+    allow(Datadog).to receive(:logger).and_return(logger)
+  end
+
+  context 'self' do
+    it 'detects if the WAF is unavailable' do
+      hide_const('Datadog::AppSec::WAF')
+
+      expect(described_class.libddwaf_required?).to be false
+    end
+
+    it 'detects if the WAF is available' do
+      stub_const('Datadog::AppSec::WAF', Module.new)
+
+      expect(described_class.libddwaf_required?).to be true
+    end
+
+    it 'reports via return of libddwaf loading failure' do
+      allow(Object).to receive(:require).with('libddwaf').and_raise(LoadError)
+
+      expect(described_class.require_libddwaf).to be false
+    end
+
+    it 'reports via return of libddwaf loading success (first require)' do
+      allow(Object).to receive(:require).with('libddwaf').and_return(true)
+
+      expect(described_class.require_libddwaf).to be true
+    end
+
+    it 'reports via return of libddwaf loading success (second require)' do
+      allow(Object).to receive(:require).with('libddwaf').and_return(false)
+
+      expect(described_class.require_libddwaf).to be true
+    end
+  end
+
+  describe '#load_libddwaf' do
+    context 'when LoadError is raised' do
+      before do
+        allow(Object).to receive(:require).with('libddwaf').and_raise(LoadError)
+      end
+
+      it { expect(described_class.new.send(:load_libddwaf)).to be false }
+    end
+
+    context 'when loaded but missing mandatory const' do
+      before do
+        allow(Object).to receive(:require).with('libddwaf').and_return(true)
+        hide_const('Datadog::AppSec::WAF')
+      end
+
+      it { expect(described_class.new.send(:load_libddwaf)).to be false }
+    end
+
+    context 'when loaded successfully' do
+      before do
+        allow(Object).to receive(:require).with('libddwaf').and_return(true)
+        stub_const('Datadog::AppSec::WAF', Module.new)
+      end
+
+      it { expect(described_class.new.send(:load_libddwaf)).to be true }
+    end
+  end
+
+  describe '#load_ruleset' do
+    before do
+      allow(Datadog::AppSec.settings).to receive(:ruleset).and_return(ruleset)
+    end
+
+    let(:basic_ruleset) do
+      {
+        'version' => '1.0',
+        'events' => [
+          {
+            'id' => 1,
+            'name' => 'Rule 1',
+            'tags' => { 'type' => 'flow1' },
+            'conditions' => [
+              { 'operation' => 'match_regex', 'parameters' => { 'inputs' => ['value2'], 'regex' => 'rule1' } },
+            ],
+            'action' => 'record',
+          }
+        ]
+      }
+    end
+
+    context 'when ruleset is default' do
+      let(:ruleset) { :recommended }
+
+      it { expect(described_class.new.send(:load_ruleset)).to be true }
+    end
```

Comment:
Minor: Should this check that `Datadog::AppSec::Assets.waf_rules` gets correctly called with the argument?